staker/block_validator.go

```diff
@@ -52,25 +50,55 @@ var (
 	validatorMsgCountLastValidationSentGauge = metrics.NewRegisteredGauge("arb/validator/msg_count_last_validation_sent", nil)
 )
 
+type LegacyValidatorInstance interface {
+	CheckLegacyValid() error
+	LegacyLastValidatedInfo() (*legacyLastBlockValidatedDbInfo, error)
+	SetLegacyValidInfo(info *legacyLastBlockValidatedDbInfo)
+}
+
+// ValidatorInstance is an interface used by BlockValidator to carry out block and MEL-state validations. BlockValidatorInstance implements
+// this interface to carry out block validations and MELValidatorInstance (TODO) implements this to carry out MEL-state validation
+type ValidatorInstance interface {
+	// Methods corresponding to ValidatorInstance's overall context
+	IsValidatedGlobalStateNew(gs validator.GoGlobalState) bool
+	LastValidatedGlobalState() validator.GoGlobalState
+	SetLastValidatedGlobalState(gs validator.GoGlobalState)
+	LastValidatedCount() (count uint64, isCaughtup bool, err error)                     // returns last validated count and a boolean indicating if the chain has caught up to that count
```

Comment:
nitpick: func name and return var names are already good enough to understand what the func does 🙂, so comment is almost redundant. 
Only renaming isCaughtup to isChainCaughtupToLastValidateCount would make the comment completely redundant.